ci-search: set fs group for mounted volumes

fgimenez opened the pull request:
The pods were crashlooping:
```
$ kubectl logs search-0 -n ci-search
I0112 12:29:59.483273       1 reaper_linux.go:59] Launching periodic reaper
I0112 12:29:59.483460       1 reaper_linux.go:69] Reaped: []
F0112 12:29:59.483357       1 main.go:57] error: unable to create directory for artifact: mkdir /var/lib/ci-search/bugs: permission denied
```
Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

rmohr commented on github/ci/services/ci-search/patches/statefulset.yaml:
```diff
@@ -28,6 +28,8 @@ spec:
         - --bugzilla-search=OPEN version:4.7,4.6,4.5,4.4,4.3,4.2 product="Container Native Virtualization (CNV)" AND delta_ts>-2w
         - --bugzilla-token-file=/etc/bugzilla/api
+      securityContext:
+        fsGroup: 1000
```
That does not look like running as root.

fgimenez replied:
Indeed, running as root was the first approach, but also works just by setting the file system group and letting it to run as user 1000, I've already updated the PR title and commit message.

fgimenez commented:
/hold

Anyway i'd like to update the tests to catch this issue, and to actually run, they are not being triggered.

rmohr commented:
/lgtm

fgimenez commented:
/rehearse

fgimenez commented:
/unhold

Added a few changes:
* updated deck URL to the current instead of prow.kubevirt.io
*  using latest bootstrap image with bazelisk installed
* make service tests required

rmohr commented:
/rehearse

rmohr commented:
/lgtm
/approve
